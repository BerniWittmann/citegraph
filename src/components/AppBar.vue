<template>
  <v-app-bar
    app
    dark
    :clipped-left="breakPoint"
  >
    <v-app-bar-nav-icon @click.stop="toggleDrawer"></v-app-bar-nav-icon>

    <v-toolbar-title @click="$router.push('/')">{{ $t('citegraph') }}</v-toolbar-title>

    <template v-slot:extension>
      <v-tabs
        align-with-title
        background-color="transparent"
      >
        <v-tab>Tab 1</v-tab>
        <v-tab>Tab 2</v-tab>
        <v-tab>Tab 3</v-tab>
      </v-tabs>
    </template>
  </v-app-bar>
</template>
<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'

@Component
export default class AppBar extends Vue {
  get breakPoint (): boolean {
    return this.$vuetify.breakpoint.lgAndUp
  }

  toggleDrawer () {
    this.$emit('toggle-drawer')
  }
}
</script>

<style scoped lang="scss">
</style>
