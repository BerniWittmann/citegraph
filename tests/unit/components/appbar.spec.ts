import { RouterLinkStub, mount } from '@vue/test-utils'
import { i18n, vuetify } from '../setupPlugins'

import AppBar from '@/components/AppBar.vue'

describe('components/AppBar.vue', () => {
  const router = {
    push: jest.fn().mockResolvedValue(undefined),
    resolve: jest.fn()
  }
  const getWrapper = () => {
    return mount(AppBar, {
      i18n,
      vuetify,
      mocks: {
        $route: {
          name: 'home'
        },
        $vuetify: {
          breakpoint: {
            lgAndUp: true
          }
        },
        $router: router
      },
      stubs: {
        RouterLink: RouterLinkStub
      },
      sync: false
    })
  }
  it('renders', () => {
    const wrapper = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('navigates to the home page when clicking on the title', () => {
    const wrapper = getWrapper()
    expect(wrapper.find('.v-toolbar__title').exists()).toBeTruthy()
    wrapper.find('.v-toolbar__title').trigger('click')
    expect(router.push).toHaveBeenCalledWith('/')
  })
  it('emits an event when the icon is clicked', () => {
    const wrapper = getWrapper()
    expect(wrapper.emitted('toggle-drawer')).toBeFalsy()
    expect(wrapper.find('.v-app-bar__nav-icon').exists()).toBeTruthy()
    wrapper.find('.v-app-bar__nav-icon').trigger('click')
    expect(wrapper.emitted('toggle-drawer')).toBeTruthy()
  })
})
