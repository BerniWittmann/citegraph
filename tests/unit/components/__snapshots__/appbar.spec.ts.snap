// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AppBar.vue renders 1`] = `
<header class="v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--extended v-app-bar v-app-bar--fixed" style="height: 112px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
  <div class="v-toolbar__content" style="height: 64px;"><button type="button" class="v-app-bar__nav-icon v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu theme--dark"></i></span></button>
    <div class="v-toolbar__title">citegraph</div>
  </div>
  <div class="v-toolbar__extension" style="height: 48px;">
    <div class="v-tabs v-tabs--align-with-title theme--dark">
      <div role="tablist" class="v-item-group theme--dark v-slide-group v-tabs-bar v-tabs-bar--is-mobile white--text transparent">
        <div class="v-slide-group__prev v-slide-group__prev--disabled">
          <!---->
        </div>
        <div class="v-slide-group__wrapper">
          <div class="v-slide-group__content v-tabs-bar__content">
            <div class="v-tabs-slider-wrapper" style="transition: none;">
              <div class="v-tabs-slider"></div>
            </div>
            <div tabindex="0" aria-selected="true" role="tab" class="v-tab v-tab--active">Tab 1</div>
            <div tabindex="0" aria-selected="false" role="tab" class="v-tab">Tab 2</div>
            <div tabindex="0" aria-selected="false" role="tab" class="v-tab">Tab 3</div>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
